tango/format.py

```diff
@@ -346,3 +349,41 @@ def __next__(self) -> T:
             self.f.close()
             self.f = None
             raise StopIteration()
+
+
+@Format.register("sqlite")
+class SqliteDictFormat(Format[DatasetDict]):
+    VERSION = 3
+
+    def write(self, artifact: DatasetDict, dir: Union[str, PathLike]):
+        dir = pathlib.Path(dir)
+        with gzip.open(dir / "metadata.dill.gz", "wb") as f:
+            dill.dump(artifact.metadata, f)
+        for split_name, split in artifact.splits.items():
+            filename = f"{split_name}.sqlite"
+            if not filename_is_safe(filename):
+                raise ValueError(f"{split_name} is not a valid name for a split.")
+            try:
+                (dir / filename).unlink()
+            except FileNotFoundError:
+                pass
+            if isinstance(split, SqliteSparseSequence):
+                split.copy_to(dir / filename)
+            else:
+                sqlite = SqliteSparseSequence(dir / filename)
+                sqlite.extend(split)
+
+    def read(self, dir: Union[str, PathLike]) -> DatasetDict:
+        dir = pathlib.Path(dir)
+        with gzip.open(dir / "metadata.dill.gz", "rb") as f:
+            metadata = dill.load(f)
+        splits = {
+            filename.stem: SqliteSparseSequence(filename, read_only=True)
+            for filename in dir.glob("*.sqlite")
+        }
+        return DatasetDict(metadata=metadata, splits=splits)
+
+    def checksum(self, dir: PathOrStr) -> str:
+        # This is not trivial to implement because sqlite files can be different even if they contain the same
+        # data.
+        raise NotImplementedError()
```

Comment:
This makes this format unusable. Maybe we just change the signature of `checksum()` to return `Optional[str]`, and handle that in `StepCache`?

Reply:
That's fine with me

Reply:
e54e31f